addons/l10n_id_efaktur/tests/test_l10n_id_efaktur.py

```diff
@@ -277,3 +279,304 @@ def test_efaktur_consume_code(self):
+
+        self.assertEqual(self.efaktur.available, 10)
+
+        # Generate 8 invoices and reset the last invoice, should only add availability without
+        # affecting the range
+        invoices = self.env["account.move"].create([{
+            "move_type": "out_invoice",
+            "partner_id": self.partner_id.id,
+            "invoice_date": "2019-05-01",
+            "date": "2019-05-01",
+            "invoice_line_ids": [
+                Command.create({"name": "line1", "price_unit": 110.0, "tax_ids": self.tax_id.ids}),
+            ],
+            "l10n_id_kode_transaksi": "01",
+        } for i in range(8)])
+        invoices.action_post()
+        self.assertEqual(invoices[-1].l10n_id_tax_number, '0100000000000008')
+        self.assertEqual(self.efaktur.available, 2)
+
+        last_inv = invoices[-1]
+        last_inv.button_draft()
+        last_inv.button_cancel()
+        last_inv.reset_efaktur()
+
+        self.assertEqual(self.efaktur.min, '0000000000001')
+        self.assertEqual(self.efaktur.max, '0000000000010')
+        self.assertEqual(self.efaktur.available, 3)
+
+    def test_efaktur_release_number_not_last(self):
+        """ Ensure when the number returned is not last released number, split the efaktur range into 2 parts
+        range: 1-10, used up to 2, then release 1 => split range into [1-1] available 1 & [2-10] available 8
+        """
+        out_invoice_1 = self.env["account.move"].create({
+            "move_type": "out_invoice",
+            "partner_id": self.partner_id.id,
+            "invoice_date": "2019-05-01",
+            "date": "2019-05-01",
+            "invoice_line_ids": [
+                Command.create({"name": "line1", "price_unit": 110.0, "tax_ids": self.tax_id.ids}),
+            ],
+            "l10n_id_kode_transaksi": "01",
+        })
+        out_invoice_2 = self.env["account.move"].create({
+            "move_type": "out_invoice",
+            "partner_id": self.partner_id.id,
+            "invoice_date": "2019-05-01",
+            "date": "2019-05-01",
+            "invoice_line_ids": [
+                Command.create({"name": "line1", "price_unit": 110.0, "tax_ids": self.tax_id.ids}),
+            ],
+            "l10n_id_kode_transaksi": "01",
+        })
+        out_invoice_1.action_post()
+        out_invoice_2.action_post()
+
+        out_invoice_1.button_draft()
+        out_invoice_1.button_cancel()
+        out_invoice_1.reset_efaktur()
+
+        self.assertEqual(self.efaktur.min, '0000000000001')
+        self.assertEqual(self.efaktur.max, '0000000000001')
+        self.assertEqual(self.efaktur.available, 1)
+
+        other_range = self.env['l10n_id_efaktur.efaktur.range'].search([('id', '!=', self.efaktur.id)], limit=1)
+
+        self.assertEqual(other_range.min, '0000000000002')
+        self.assertEqual(other_range.max, '0000000000010')
+        self.assertEqual(other_range.available, 8)
+
+    def test_efaktur_change_max(self):
```

Comment:
This test is there twice :thinking: 

test_change_min_max

Could you merge in a single complete test? thanks